### PR TITLE
Skip tests properly

### PR DESCRIPTION
**Pull request by amyeroberts**
# What does this PR do?

Makes sure all tests which are skipped are done so with an explicit skip. This helps with test stats, and doesn't make us think some tests are passing when they're actually empty which can be confusing c.f. [this comment](https://github.com/huggingface/transformers/pull/31495#discussion_r1653085411)

### CODE REVIEW

File: tests/deepspeed/test_deepspeed.py [hunk 1]
```diff
@@ -643,7 +643,7 @@ def test_early_get_last_lr(self, stage, dtype):
             # print(trainer.model.b.item())
             # need to investigate at some point
             if (stage == ZERO3 and dtype == FP16) or (dtype == BF16):
-                return
+                self.skipTest("When using zero3/fp16 or any/bf16 the optimizer seems run oddly")
```
Comment (ydshieh) [inline tests/deepspeed/test_deepspeed.py:646]: should we want to add `reason=` here too (and other similar places)?

**Review by ydshieh — APPROVED** [review]
Wow, big change! Thank you for taking care of the library!

File: tests/deepspeed/test_deepspeed.py (hunk 1 above)
Comment (amyeroberts) [reply tests/deepspeed/test_deepspeed.py:646]: Sure! I can add that

**Review by ArthurZucker — COMMENTED** [review]
thanks for being thourough

**Pull request merged**